File: 75_sort_colors.js
var sortColors = function (nums) {
  let index = 0;
  let n = nums.length;

  // Move all 0s to the front
  for (let i = 0; i < n; i++) {
    if (nums[i] === 0) {
      [nums[i], nums[index]] = [nums[index], nums[i]];
      index++;
    }
  }

  // Move all 1s to the correct position
  for (let i = index; i < n; i++) {
    if (nums[i] === 1) {
      [nums[i], nums[index]] = [nums[index], nums[i]];
      index++;
    }
  }

  // Remaining elements are 2s, so they are already in place
};

File: 75_sort_colors.py
from typing import List

class Solution:
    def sortColors(self, nums: List[int]) -> None:
        n = len(nums)
        index = 0

        # Move all 0s to the front
        for i in range(n):
            if nums[i] == 0:
                nums[i], nums[index] = nums[index], nums[i]
                index += 1

        # Move all 1s to the correct position
        for i in range(index, n):
            if nums[i] == 1:
                nums[i], nums[index] = nums[index], nums[i]
                index += 1

        # Move all 2s to the correct position (remaining part is already sorted)
